cartographer/mapping_3d/submaps.cc

```diff
@@ -226,6 +226,22 @@ void Submap::ToProto(mapping::proto::Submap* const proto) const {
       low_resolution_hybrid_grid().ToProto();
 }
 
+void Submap::UpdateFromProto(const mapping::proto::Submap& proto) {
+  CHECK(proto.has_submap_3d());
+  const auto& submap3d = proto.submap_3d();
+  SetNumRangeData(submap3d.num_range_data());
+  finished_ = submap3d.finished();
+  high_resolution_hybrid_grid_ =
```

Comment:
This behavior is inconsistent with 2D, where the grid is unchanged if the proto does not contain one.

Reply:
Thanks! Done!